apps/functions-subscription/src/adapters/azure/functions/__tests__/list-trials.test.ts

```diff
@@ -0,0 +1,149 @@
+
+    const actual = await makeListTrialsHandler(env)(
+      makeAValidListTrialRequest({ pageSize: 'aString' }),
+      makeFunctionContext(),
+    );
+
+    expect(actual.status).toStrictEqual(400);
+    expect(await actual.json()).toMatchObject({
+      status: 400,
+      title: 'Bad Request',
+    });
+  });
+
+  it('should return 403 if x-user-groups header does not contain the correct group', async () => {
+    const request = new HttpRequest({
+      url: makeAValidListTrialRequest().url,
+      method: makeAValidListTrialRequest().method,
+      headers: {
+        ...supportHttpRequestHeaders,
+        'x-user-groups': 'Guest,AnotherGroup',
+      },
+    });
+    const env = makeTestSystemEnv();
+    const actual = await makeListTrialsHandler(env)(
+      request,
+      makeFunctionContext(),
+    );
+    expect(actual.status).toStrictEqual(403);
+    expect(await actual.json()).toMatchObject({
+      status: 403,
+    });
+  });
+
+  it('should return 500 when an error occurred', async () => {
+    const env = makeTestSystemEnv();
+
+    const error = new Error('Something went wrong');
+    env.listTrials.mockReturnValueOnce(TE.left(error));
+
+    const actual = await makeListTrialsHandler(env)(
+      makeAValidListTrialRequest(),
+      makeFunctionContext(),
+    );
+
+    expect(actual.status).toStrictEqual(500);
+    expect(await actual.json()).toMatchObject({
+      status: 500,
+      detail: error.message,
+    });
+  });
+
+  it('should return 200 when trials exists', async () => {
+    const env = makeTestSystemEnv();
+
+    const aSlimTrial = {
+      id: aTrial.id,
+      name: aTrial.name,
+      state: aTrial.state,
+      description: aTrial.description,
+    } as Trial;
+
+    const anotherSlimTrial = {
+      ...aSlimTrial,
+      id: 'anotherTrialId' as TrialId,
+      name: 'anotherTrialName' as NonEmptyString,
+    } as Trial;
+
+    env.listTrials.mockReturnValueOnce(TE.right([aTrial, anotherTrial]));
+
+    const actual = await makeListTrialsHandler(env)(
+      makeAValidListTrialRequest({
+        pageSize: '2',
+        maximumId: 'z',
+        minimumId: 'a',
+      }),
+      makeFunctionContext(),
+    );
+
+    expect(env.listTrials).nthCalledWith(1, {
+      pageSize: 2,
+      maximumId: 'z',
+      minimumId: 'a',
+    });
+
+    const result = await actual.json();
+
+    expect(actual.status).toStrictEqual(200);
+
+    expect(result).toMatchObject({
+      items: [aSlimTrial, anotherSlimTrial],
+      previousId: anotherSlimTrial.id,
+      nextId: aSlimTrial.id,
+    });
+  });
+
+  it('should return empty list when the are no trials', async () => {
+    const env = makeTestSystemEnv();
+
+    env.listTrials.mockReturnValueOnce(TE.right([]));
+
+    const actual = await makeListTrialsHandler(env)(
+      makeAValidListTrialRequest(),
+      makeFunctionContext(),
+    );
+
+    expect(env.listTrials).nthCalledWith(1, {
+      pageSize: 25,
+      maximumId: undefined,
+      minimumId: undefined,
```

Comment:
**suggestion**: i think you can omit those params

```suggestion
```

Reply:
Removed in ce70af07d12fba443dabfbc5aa2b2a899a27949b